header.go

```diff
@@ -341,9 +394,9 @@ func (h *Header) Create(filename string) (string, error) {
 
 // Update updates an existing license header if it matches the
 func (h *Header) Update(filename, header string) (string, bool, error) {
-	cs, err := detectCommentStyle(header)
-	if err == nil {
-		header = cs.Parse(header)
+	header, cs, err := parseComment(header)
+	if err != nil {
+		return "", false, fmt.Errorf("parse header comment: %w", err)
```

Comment:
I want to avoid failing to lint as much as possible, so instead of failing, if we can't parse the comment I think it makes sense to either ignore the file or generate a new header instead.